AmplifyPlugins/Core/AWSPluginsCoreTests/ServiceConfiguration/AmplifyAWSServiceConfigurationTests.swift

```diff
@@ -26,4 +39,46 @@ class AmplifyAWSServiceConfigurationTests: XCTestCase {
         XCTAssertEqual(expectedSystem, userAgentParts[1])
         XCTAssertEqual(expectedLocale, userAgentParts[2])
     }
+
+    /// Test adding a new platform to AmplifyAWSServiceConfiguration
+    ///
+    /// - Given: Amplify library
+    /// - When:
+    ///    - I add a new platform to the AmplifyAWSServiceConfiguration
+    /// - Then:
+    ///    - AmplifyAWSServiceConfiguration should be configured properly with the new platform added.
+    ///
+    func testAddNewPlatform() {
+        AmplifyAWSServiceConfiguration.addPlatform(.flutter, version: "1.1")
+        let currentSystemName = UIDevice.current.systemName.replacingOccurrences(of: " ", with: "-")
+        let currentSystemVersion = UIDevice.current.systemVersion
+        let expectedLocale = Locale.current.identifier
+        let expectedSystem = "\(currentSystemName)/\(currentSystemVersion)"
+
+        let configuration = AmplifyAWSServiceConfiguration()
+
+        XCTAssertNotNil(configuration.userAgent)
+        let userAgentParts = configuration.userAgent.components(separatedBy: " ")
+        XCTAssertEqual(4, userAgentParts.count)
+        XCTAssert(configuration.userAgent.contains(AmplifyAWSServiceConfiguration.Platform.flutter.rawValue))
+        XCTAssertEqual(expectedSystem, userAgentParts[2])
+        XCTAssertEqual(expectedLocale, userAgentParts[3])
+    }
+
+    /// Test overriding iOS platform in AmplifyAWSServiceConfiguration
+    ///
+    /// - Given: Amplify library
+    /// - When:
+    ///    - I override iOS platform with a new version
```

Comment:
A "new" version, or "any" version? If the latter, why do we even expose `.ios` as an enum case?

Reply:
Currently `.ios` case can only be used internally and we ignore the value set by the external developer. This make sure that iOS value is not overriden by developers. There is no need to expose `.ios` but I dont think we can have private enum cases. 

I can make the function that create the platform string like shown below and remove `ios` case:

```swift
    static func platformInformation() -> String {
         let additionalPlatform = platformMapping.map {"\($0.rawValue)/\($1)"}.joined(separator: " ")

         // add iOS here
         let version = "1.1.0"
         let sdkName = "amplify-iOS"
         return "\(sdkName)/\(version) \(additionalPlatform)"
     }
```

Reply:
Yes, this makes sense